### PR TITLE
add crypto_box_seed_keypair

### PR DESCRIPTION
**Pull request by DiogoAbu**
This will add `crypto_box_seed_keypair` that allows to derive a key pair from a seed key.
Fix typo on types;
Add new type relative to the function added.

Edit:
I see my prettier extension formatted the type declaration file, hope that's okay.
And i haven't tested it on IOS.

### CODE REVIEW

**Comment by lyubo** [comment]
Hey @DiogoAbu , thanks for this PR. 
What's the reason of removing the checks like this one?

```
      if (Sodium.crypto_secretbox_keygen(key) != 0)
        p.reject(ESODIUM,ERR_FAILURE);
```

**Comment by DiogoAbu** [comment]
Hello!
I didn't remove it though... was it ever there? I mean i wrote the `crypto_secretbox_keygen` method, and I can't find the check anywhere on the file history.
Should I add it?

Edit.
Ok, I see what you mean.
I set-up this repo as a upstream on my fork, updated the fork, and applied the changes that you see on this pull request.
This PR is showing 7 commits, but in reality only the last commit is new code.

File: android/src/main/java/org/libsodium/rn/RCTSodiumModule.java [hunk 1]
```diff
@@ -127,14 +127,11 @@ public void randombytes_stir(final Promise p) {
   public void crypto_secretbox_keygen(final Promise p){
     try {
       byte[] key = new byte[Sodium.crypto_secretbox_keybytes()];
-
-      if (Sodium.crypto_secretbox_keygen(key) != 0)
```
Comment (DiogoAbu) [inline android/src/main/java/org/libsodium/rn/RCTSodiumModule.java]: This code is part of the other pull request I made.

**Review by lyubo — CHANGES_REQUESTED** [review]
Please, split crypto_box_seed_keypair and crypto_pwhash* to separate PRs. Also, do not commit white-space changes to code which are not relevant to new functionality.

**Comment by lyubo** [comment]
Closing this one because of conflicts with master branch. Please submit a new one.

**Pull request closed without merging**